flask_appbuilder/validators.py

```diff
@@ -31,3 +51,41 @@ def __call__(self, form, field):
                 if self.message is None:
                     self.message = field.gettext(u"Already exists.")
                 raise ValidationError(self.message)
+
+
+class PasswordComplexityValidator:
+    """
+    WTForm field validator. Calls a custom password validator, useful for imposing
+    password complexity for database Auth users.
+    """
+
+    def __call__(self, form: Form, field: Field) -> None:
+        password_complexity_validator = current_app.config.get(
+            "FAB_PASSWORD_COMPLEXITY_VALIDATOR", None
+        )
+        if password_complexity_validator is not None:
+            try:
+                password_complexity_validator(field.data)
+            except PasswordComplexityValidationError as exc:
+                raise ValidationError(str(exc))
+        if current_app.config.get("FAB_PASSWORD_COMPLEXITY_ENABLED", False):
```

Comment:
I think the current set of configuration variables is a bit confusing.

When I see `FAB_PASSWORD_COMPLEXITY_ENABLED` - think this parameter configures password complexity check in general, so I expect that if `FAB_PASSWORD_COMPLEXITY_ENABLED` is "false" then no validation is done (even if I have my own custom validator configured in `FAB_PASSWORD_COMPLEXITY_VALIDATOR`).

Currently, in order to enable ONLY custom validator you have to have effectively this combination:

```
FAB_PASSWORD_COMPLEXITY_ENABLED = false
FAB_PASSWORD_COMPLEXITY_VALIDATOR = your_validator
```

This is pretty confusing (because I'd expect `FAB_PASSWORD_COMPLEXITY_ENABLED = false` disables validation of password completely.

I think two one of the two approaches would be better:

a) `FAB_PASSWORD_COMPLEXITY_ENABLED` - enables validation in general, and `FAB_PASSWORD_COMPLEXITY_VALIDATOR` replaces the default validator with the custom one. In this case you cannot run both custom and default validation at the same time

b) `FAB_PASSWORD_COMPLEXITY_ENABLED` - should be renamed to `FAB_PASSWORD_DEFAULT_COMPLEXITY_VALIDATOR` - to not make impression that whole feature is disabled by setting it to `false`

I have slight preference for option a) (I think it's better to always run one validator - either default or custom, you can always derive your custom validator from the default one and use it internally) - but I think both a) and b) are better than current proposal.

Reply:
Very good point, went for option a)

Reply:
Cool. Looks great now. Then I have one more NIT comment :)